### ISSUE TITLE
Re-bound Keyboard Shortcuts do not update in menus

### ISSUE THREAD

**Issue by GCRev** [rstudio/rstudio#9915]
### System details
```
    RStudio Edition : Desktop/Server
    RStudio Version : Prairie Trillium Daily (50eb093bd204)
    OS Version      : Ubuntu 21.04 Hirsute
    R Version       : 4.1.1
```
### Steps to reproduce the problem

This behavior is described in https://github.com/rstudio/rstudio/issues/5256, but this issue is focused **only on the menus.**
1. Open the Keyboard Shortcut Editor
2. Change two commands **that appear in menus** (not all commands do): one with a default keyboard shortcut and one without
3. Verify that neither command as it appears in menus shows the updated shortcut.

https://github.com/rstudio/rstudio/issues/5256 Illustrates an example of this with images. In this example the user has re-bound the shortcut of the `Build Package Documentation` command to `Shift+Cmd+D` from the default `Ctrl+Shift+D`. The menus still show the default value of `Ctrl+Shift+D`.
![Shortcut editor](https://user-images.githubusercontent.com/111939/135112521-59c2c68d-b10b-4a72-b754-47e1bb4115cc.png)
![Menu example](https://user-images.githubusercontent.com/111939/135112544-847a18c8-0eb9-4936-ab23-4c7771a66b22.png)

### Describe the problem in detail

Custom shortcuts are not displayed properly in application menus when the user re-assigns an existing command, or when a user assigns a shortcut to a previously-unassigned command. This leads to confusion, in some cases, because the menus will always display the default values of the commands, rather than their updated values.

The shortcuts function normally, however, and the user can use them as they would expect. They also appear correctly in the command palette once the user changes them from the settings.

### Describe the behavior you expected

After making changes to a command's shortcut value, this value should be accurately reflected in any menus in which the commands appear. This includes overriding default shortcuts with new ones, or even clearing default shortcuts out entirely.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] ~If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).~
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by astayleraz** [comment]
Verified in 2022.01.1-daily-171.pro1 I was able to re-bound keyboard shortcuts and the drop-down menu would show the changes as well as the keyboard shortcut help pop up. Also verified the changed shortcuts new command would work as expected and added an automation test to cover these cases. Closing issue.
